### PR TITLE
Implement CLI/REST server support for new messages

### LINKED ISSUE #119: Add REST server support for new messages
Add support for `MsgUpgradeAdminsitrator` and `MsgMigrateContract` to `client/rest`. Also update the return value of `ContractInfo` to contain the `Administrator`.

Please put this code in a separate file as we will need to update it when merging into the cosmos-sdk 0.39-master branch.

### LINKED ISSUE #120: Add CLI client support for new messages
Add support for `MsgUpgradeAdminsitrator` and `MsgMigrateContract` to `client/cli`. Also update the return value of `ContractInfo` to contain the `Administrator`.

Please put this code in a separate file as we will need to update it when merging into the cosmos-sdk 0.39-master branch.

### PR DESCRIPTION
**Pull request by alpe**
🚧 After #124

* Resolves #119
* Resolves #120

CLI/REST Support:
* Set/clear admin on contract
* Migrate contract to new code id

## Reviewer note
- No tests implemented, yet 🚨 
- Rest routes are currently discussed 🗣️ 

--

- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by alpe** [comment]
I did some manual tests with a local instance:
```sh
wasmcli keys add fred
wasmcli tx send $(wasmcli keys show validator -a) $(wasmcli keys show fred -a) 5000000ucosm --chain-id=testing
# store first version
wasmcli tx wasm store contract.wasm --from validator --gas 600000  -y --chain-id=testing
wasmcli query wasm list-code

# create instance with admin: fred
wasmcli tx wasm instantiate 1 '{"verifier":"cosmos1rsj2m3qas3jy50s8psns95qka09lhwhx2wvdq8","beneficiary":"cosmos1rsj2m3qas3jy50s8psns95qka09lhwhx2wvdq8"}'  --from validator --amount=50000ucosm  --label "escrow 1" -y --chain-id=testing --admin=$(wasmcli keys show fred -a)

# migrate with user validator => fails
wasmcli tx wasm migrate cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5 2 "{}" --from validator --chain-id=testing


# migrate with user fred => ok
wasmcli tx wasm migrate cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5 2 "{}" --from fred --chain-id=testing


# show status
wasmcli query wasm contract cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5

{
  "code_id": 2,
  "creator": "cosmos1r5kgu6y7fqfh4dksw4c28hv7ns9ydnusf33vgs",
  "admin": "cosmos1rsj2m3qas3jy50s8psns95qka09lhwhx2wvdq8",
  "label": "escrow 1",
  "init_msg": {
    "verifier": "cosmos1rsj2m3qas3jy50s8psns95qka09lhwhx2wvdq8",
    "beneficiary": "cosmos1rsj2m3qas3jy50s8psns95qka09lhwhx2wvdq8"
  },
  "address": "cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5"
}

```

File: x/wasm/client/cli/tx.go [hunk 1]
```diff
@@ -131,20 +135,31 @@ func InstantiateContractCmd(cdc *codec.Codec) *cobra.Command {
 
 			initMsg := args[1]
 
+			adminStr := viper.GetString(flagAdmin)
```
Comment (ethanfrey) [inline x/wasm/client/cli/tx.go:138]: nice

File: x/wasm/client/cli/new_tx.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	}
+	return cmd
+}
+
+// UpdateContractAdminCmd sets or clears an admin for a contract
+func UpdateContractAdminCmd(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set-contract-admin [contract_addr_bech32] [optional_new_admin_addr_bech32]",
+		Short: "Set new admin for a contract. Can be empty to prevent further migrations",
+		Args:  cobra.RangeArgs(1, 2),
```
Comment (ethanfrey) [inline x/wasm/client/cli/new_tx.go:58]: i didn't know this one. I like it.

i kind of think setting it to `nil` should require a bit more intention than a missing argument. but cli ux is definitely not high on my list of concerns (or anyone's I think).

Maybe a flag for this? Fine to leave as is.

**Comment by ethanfrey** [comment]
If you add the field to query and manually test, feel free to merge without another review

**Comment by ethanfrey** [comment]
Sorry @alpe, I missed that the field was there. I expected it needed a code change and didn't come automatically. 

Good to merge as is

File: x/wasm/client/cli/new_tx.go (hunk 2 above)
Comment (alpe) [reply x/wasm/client/cli/new_tx.go:58]: 👍 A flag is a good idea. Will add it.

Comment (alpe) [reply x/wasm/client/cli/new_tx.go:60]: Added `--no-admin`

**Comment by alpe** [comment]
Manual CLI tests for set new  contract admin
```
wasmcli keys add greg
wasmcli tx send $(wasmcli keys show validator -a) $(wasmcli keys show greg -a) 5000000ucosm --chain-id=testing

# non admin -> fail
wasmcli tx wasm set-contract-admin cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5  $(wasmcli keys show greg -a) --from validator --chain-id=testing

# as admin -> success
wasmcli tx wasm set-contract-admin cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5  $(wasmcli keys show greg -a) --from fred --chain-id=testing
# confirm
wasmcli query wasm contract cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5

# fred not admin -> fail
wasmcli tx wasm set-contract-admin cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5  $(wasmcli keys show greg -a) --from fred --chain-id=testing

# clear admin
wasmcli tx wasm set-contract-admin cosmos18vd8fpwxzck93qlwghaj6arh4p7c5n89uzcee5  --no-admin --from greg --chain-id=testing -y -b=block
```

**Pull request merged**